packages/event-handler/src/rest/BaseRouter.ts

```diff
@@ -0,0 +1,155 @@
+import type { GenericLogger } from '@aws-lambda-powertools/commons/types';
+import { isRecord } from '@aws-lambda-powertools/commons/typeutils';
+import {
+  getStringFromEnv,
+  isDevMode,
+} from '@aws-lambda-powertools/commons/utils/env';
+import type { Context } from 'aws-lambda';
+import type { ResolveOptions } from '../types/index.js';
+import type {
+  RouteHandler,
+  RouteOptions,
+  RouterOptions,
+} from '../types/rest.js';
+
+const HttpVerbs = [
+  'GET',
+  'POST',
+  'PUT',
+  'PATCH',
+  'DELETE',
+  'HEAD',
+  'OPTIONS',
+] as const;
+
+abstract class BaseRouter {
+  protected context: Record<string, unknown>; // TODO: should this be a map instead?
+  /**
+   * A logger instance to be used for logging debug, warning, and error messages.
+   *
+   * When no logger is provided, we'll only log warnings and errors using the global `console` object.
+   */
+  protected readonly logger: Pick<GenericLogger, 'debug' | 'warn' | 'error'>;
+  /**
+   * Whether the router is running in development mode.
+   */
+  protected readonly isDev: boolean = false;
+
+  public constructor(options?: RouterOptions) {
+    this.context = {};
+    const alcLogLevel = getStringFromEnv({
+      key: 'AWS_LAMBDA_LOG_LEVEL',
+      defaultValue: '',
+    });
+    this.logger = options?.logger ?? {
+      debug: alcLogLevel === 'DEBUG' ? console.debug : () => undefined,
+      error: console.error,
+      warn: console.warn,
+    };
+    this.isDev = isDevMode();
+  }
+
+  public abstract resolve(
+    event: unknown,
+    context: Context,
+    options?: ResolveOptions
+  ): Promise<unknown>;
+
+  public abstract route(handler: RouteHandler, options: RouteOptions): void;
```

Comment:
This method will allow customers to use `@app.route("/todos", methods=["POST", "GET"]`, for example?. Don't get hung up on the name I suggested, just the experience.

This can ease the experience for those that are building on top of RESTful interface and wants to use the same resource/path for multiple HTTP verbs.

Reply:
Yes, but at this point in time it's just abstract and the implementation will be done in specific classes (i.e. `AlbResolver`).

The DX is slightly different as it looks like this but it's equivalent:

```ts
app.route(<handler>, {
  method: ["POST", "GET"],
  path: "/todos"
})
```

I'm unsure if the type for `method` accepts arrays of `HttpVerb` or not yet, but it will in future iterations.